test/unpack_visitor.cpp

```diff
@@ -0,0 +1,135 @@
+#include <msgpack.hpp>
+#include <gtest/gtest.h>
+#include <sstream>
+
+// To avoid link error
+TEST(unpack_visitor, dummy)
+{
+}
+
+#if MSGPACK_DEFAULT_API_VERSION >= 2
+
+struct json_visitor : msgpack::v2::null_visitor {
```

Comment:
Since this test is also listed on the wiki documentation page, I'll point out two potential issues:
* `end_map_value()` doesn't append a comma to separate map items like `end_array_item()` does.
* `visit_str()` doesn't process strings as Unicode, therefore non-ASCII codepoints will result in invalid JSON.

Neither of these is required by the test case so that's okay for this particular piece of code, however you might want to point it out in the documentation.